### PR TITLE
TEST: using `pytest.skip` instead of `device_filter`

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description 
* Using `pytest.skip` instead of `device_filter` when case is buggy.
* Added docstrings for `get_dataframes_and_queues`, `_convert_to_dataframe` and `get_queues` utilities.
* Backlog reviewed for tickets on TODO comment sections that were removed. All cases covered.

### TODO
* will be done in separate PR the same idea for _dataframe_filter, when case for exact dataframe lib is buggy

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [x] The unit tests pass successfully.
- [x] I have run it locally and tested the changes extensively.
- [x] I have resolved any merge conflicts that might occur with the base branch.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [x] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

File: onedal/datatypes/tests/test_data.py [hunk 1]
```diff
@@ -67,10 +68,11 @@ def _test_input_format_f_contiguous_numpy(queue, dtype):
     assert_allclose(expected, result)
 
 
-# TODO: investigate sporadic failures on GPU
 @pytest.mark.parametrize("queue", get_queues("cpu"))
```
Comment (samir-nasibli) [inline onedal/datatypes/tests/test_data.py:71]:
```suggestion
@pytest.mark.parametrize("queue", get_queues())
```

**Comment by md-shafiul-alam** [comment]
Skipping strategy looks good. What would be the best way to make sure that the to-do's are investigated and not forgotten? May be ticket?

**Comment by samir-nasibli** [comment]
> Skipping strategy looks good. What would be the best way to make sure that the to-do's are investigated and not forgotten? May be ticket?

I will check them once again before making the PR ready for review. Thank you!

**Comment by ethanglaser** [comment]
Why is this preferred? Just curious, as there is no functional difference (I think)

**Comment by samir-nasibli** [comment]
> Why is this preferred? Just curious, as there is no functional difference (I think)

Device filter is needed to create test cases for testing.
This method of test deselection occurs unnoticeably and is not visible in the logs. pytest and additional tools provide wide opportunities for collecting information about skipped, xfaild and other tests cases.

File: sklearnex/linear_model/tests/test_logreg.py [hunk 2]
```diff
@@ -37,11 +37,10 @@ def prepare_input(X, y, dataframe, queue):
...
 
 
-@pytest.mark.parametrize(
-    "dataframe,queue",
-    get_dataframes_and_queues(device_filter_="cpu"),
-)
+@pytest.mark.parametrize("dataframe,queue", get_dataframes_and_queues())
 def test_sklearnex_multiclass_classification(dataframe, queue):
+    if queue and queue.sycl_device.is_gpu:
+        pytest.skip("Logistic Regression fit for the GPU sycl_queue is buggy.")
```
Comment (samir-nasibli) [inline sklearnex/linear_model/tests/test_logreg.py:43]: @avolkov-intel is this skip comment is correct?
Or it is just not implemented yet on GPU?

Comment (ethanglaser) [reply sklearnex/linear_model/tests/test_logreg.py:43]: I think its not yet supported, only binary

**Comment by samir-nasibli** [comment]
Latest CI job: http://intel-ci.intel.com/ef38b951-68c3-f150-b2d6-a4bf010d0e2e

File: sklearnex/linear_model/tests/test_logreg.py (hunk 2 above)
Comment (samir-nasibli) [reply sklearnex/linear_model/tests/test_logreg.py:43]: Thank you @ethanglaser ! Will update then

Comment (avolkov-intel) [reply sklearnex/linear_model/tests/test_logreg.py:43]: Ethan is right, please change comment in pytest.skip

File: onedal/svm/tests/test_svc.py [hunk 3]
```diff
@@ -72,9 +73,10 @@ def test_class_weight(queue):
     assert_array_almost_equal(clf.predict(X, queue=queue), [2] * 6)
 
 
-# TODO: investigate sporadic failures on GPU
-@pytest.mark.parametrize("queue", get_queues("cpu"))
+@pytest.mark.parametrize("queue", get_queues())
 def test_sample_weight(queue):
+    if queue and queue.sycl_device.is_gpu:
+        pytest.skip("Sporadic failures on GPU sycl_queue.")
```
Comment (avolkov-intel) [inline onedal/svm/tests/test_svc.py:79]: sample weight parameter is not supported on GPU, please change comment

**Review by Alexsandruss — APPROVED** [review]
